File: programmers/lv2/괄호_회전하기.py
def is_right(s):
    st = []
    for p in s:
        if p == ')':
            if len(st) == 0  or st.pop() != '(':
                return False
        elif p == ']':
            if len(st) == 0 or st.pop() != '[':
                return False
        elif p == '}':
            if len(st) == 0 or st.pop() != '{':
                return False
        else:
            st.append(p)
    return True if len(st) == 0 else False
                
    
def solution(s):
    answer = 0
    s = [i for i in s]

    for i in range(len(s)):
        if is_right(s):
            answer += 1
        s.append(s.pop(0))
    return answer

File: programmers/lv2/괄호_회전하기.js
function solution(s) {
  let answer = 0;
  for (let i = 0; i < s.length; i++) {
    const rotated = s.slice(i) + s.slice(0, i);
    if (isCorrect(rotated)) answer += 1;
  }
  return answer;
}

function isCorrect(s) {
  const stack = [];
  const pair = { '}': '{', ']': '[', ')': '(' };
  const arr = s.split('');
  for (let i = 0; i < arr.length; i++) {
    if (['(', '{', '['].includes(arr[i])) {
      stack.push(arr[i]);
    } else {
      if (stack.pop() !== pair[arr[i]]) {
        return false;
      }
    }
  }

  return stack.length === 0;
}
